docs/user/customization.md

```diff
@@ -64,4 +64,167 @@ An example `worker.ign` is shown below. It has been modified to increase the HTT
 }
 ```
 
+## Install time customization for machine configuration
+
+**IMPORTANT**:
+
+- These customizations require using the `manifests` target that does not provide compatibility guarantees, for more information [check here](versioning.md#versioning).
+- This can affect upgradability of your cluster as the `machine-config-operator` can mark clusters tainted when user defined [MachineConfig][machine-config] objects are present in the cluster.
+
+In most cases, user applications should be run on the cluster via Kubernetes workload objects (e.g. DaemonSet, Deployment, etc). For example, DaemonSets are the most stable way to run a logging agent on all hosts. However, there may be some cases where these workloads need to be executed prior to the node joining the Kubernetes cluster. For example, a compliance mandate like "the user must run auditing tools as soon as the operating system comes up" might require a custom systemd unit for an auditing container in the Ignition config for some or all nodes.
+
+The configuration of machines in OpenShift is controlled using `MachineConfig` objects and what configuration is applied to a machine in the OpenShift cluster is based on the [MachineConfigPool][machine-config-pool] objects. To allow customization of machine configuration which is not possible as Day 2 operation, the installer allows users to bring their own custom `MachineConfig` objects.
+
+1. `openshift-install --dir $INSTALL_DIR create manifests`
+
+2. Copy files with `MachineConfig` objects to `$INSTALL_DIR/openshift/` directory.
+
+    These custom `MachineConfig` objects are black boxes to the installer and the installer only plays the role of `oc create -f <custom-machine-config-object>` early enough into cluster bootstrap to make sure the configuration is used by the [MachineConfigOperator][machine-config-operator].
+
+3. `openshift-install --dir $INSTALL_DIR create cluster`
+
+### Control plane with no Taints
+
+All control plane nodes by default register with a taint `node-role.kubernetes.io/master=:NoSchedule` making them unschedulable by most normal workloads. An imaginary installation that requires the control plane to boot without that taint can push a custom `MachineConfig` object with a `kubelet.service` that doesn't include the taint.
+
+For example on Libvirt:
```

Comment:
Should we be referencing Libvirt if that's more of an internal tool when most users will use AWS?